request.go

```diff
@@ -190,3 +206,85 @@ func (exo *Client) DetailedRequest(uri string, params string, method string, hea
 	defer response.Body.Close()
 	return exo.ParseResponse(response)
 }
+
+// prepareValues uses a profile to build a POST request
+func prepareValues(profile interface{}) (*url.Values, error) {
+	params := url.Values{}
+	value := reflect.ValueOf(profile)
+	typeof := reflect.TypeOf(profile)
+	for typeof.Kind() == reflect.Ptr {
+		typeof = typeof.Elem()
+		value = value.Elem()
+	}
```

Comment:
For my know-how, what is the intent of this loop?

Reply:
`interface{}` could be the `struct` I want to work on, but also a `*struct`, or a `**struct`, and so on.